lessonly/lldevconfig
# Standardize package management

# Description

Resolves [[ch27604]](https://app.clubhouse.com/lessonly/stories/27604)

This PR adds setup for all the basic tools needed to be an engineer at Lessonly.  It focuses on macOS, but also attempts to make everything work the same for Debian-flavored Linux OSes.

Specifically, this sets up:

- homebrew on macOS
- git
- asdf
- Generates an SSH key for Github and shows you how to add it to Github
- Installs zsh and makes it your default shell
- Installs a few necessary libraries
- Installs postgresql via Homebrew for local tools
- Runs dnsmasq on boot

Once all this is run, developers should be able to visit any Lessonly repo and run its setup scripts to get up and running quickly.

# Test notes

Start on a new dev machine. This whole process takes about a half hour.

- Run `bin/bootstrap`
- Run `bin/lldevconfig`
- Clone github.com/lessonly/lessonly and go to that project's root
    - Checkout the `armilam/ch28872/update-bin-setup-and-add-docker-compose-yml` (or `master` if that branch has already been merged)
- Run `asdf install`
- Run `bin/setup`
- Run `docker-compose up`
- Run `bundle exec rails s`
- [ ] At this point, the lessonly core app should be up and running
